src/server/server_family.cc

```diff
@@ -1763,18 +1763,39 @@ std::string ServerFamily::BuildClusterNodeReply(ConnectionContext* cntx) const {
   }
 }
 
+template <class Lock> bool try_lock_for(Lock& lock, absl::Duration duration) {
+  absl::Time start = absl::Now();
+  while (true) {
+    VLOG(1) << "Try acquire lock";
+    if (lock.try_lock()) {
+      VLOG(1) << "Acquired lock";
+      return true;
+    }
+    if (absl::Now() - start > duration) {
+      return false;
+    }
+    ThisFiber::SleepFor(200ms);
+  }
+}
+
```

Comment:
Let's not hurry with quick fixes 🙂 Boost has a timed_mutex for this, but it currently doesn't have a wrapper inside helio. Lets add one https://github.com/romange/helio/blob/master/util/fibers/fiber.h#L73